--- uncrustify-0.55/src/keywords.cpp	2009-11-20 03:15:09.000000000 +0100
+++ uncrustify-0.55-qt-foreach/src/keywords.cpp	2009-12-15 02:16:44.000000000 +0100
@@ -128,7 +128,7 @@
    { "flags",            CT_TYPE,         LANG_VALA                                                                   },
    { "float",            CT_TYPE,         LANG_ALLC                                                                   },
    { "for",              CT_FOR,          LANG_ALL                                                                    }, // PAWN
-   { "foreach",          CT_FOR,          LANG_CS | LANG_D | LANG_VALA                                                },
+   { "foreach",          CT_FOR,          LANG_CS | LANG_CPP | LANG_D | LANG_VALA                                             },
    { "foreach_reverse",  CT_FOR,          LANG_D                                                                      },
    { "forward",          CT_FORWARD,      LANG_PAWN                                                                   }, // PAWN
    { "friend",           CT_FRIEND,       LANG_CPP                                                                    },
